### PR TITLE
Ignore pods being deleted

### PR DESCRIPTION
**Pull request by uthark**
Closes https://github.com/kubernetes-sigs/descheduler/issues/618

### CODE REVIEW

**Review by damemi — COMMENTED** [review]
/ok-to-test

**Comment by uthark** [comment]
@damemi tests are passing now.

**Comment by uthark** [comment]
/retest

Previous run failed with
> ERROR: failed to create cluster: failed to pull image "kindest/node:v1.21.4": command "docker pull kindest/node:v1.21.4" failed with error: exit status 1

File: pkg/descheduler/strategies/topologyspreadconstraint_test.go [hunk 1]
```diff
@@ -3,9 +3,8 @@ package strategies
 import (
 	"context"
 	"fmt"
-	"testing"
-
 	"sigs.k8s.io/descheduler/pkg/api"
+	"testing"
```
Comment (lianghao208) [inline pkg/descheduler/strategies/topologyspreadconstraint_test.go:7]: May I know the reason why moving this from L6 to L7?

Comment (uthark) [reply pkg/descheduler/strategies/topologyspreadconstraint_test.go:6]: it was auto format in the IDE, reverted the change.

**Review by a7i — APPROVED** [review]
🎉awesome!

As a follow-up, some of the other strategies could benefit from this as well. Possibly a good place would be [ListPodsOnANode](https://github.com/kubernetes-sigs/descheduler/blob/master/pkg/descheduler/pod/pods.go#L159)), for those that use that.

File: pkg/descheduler/strategies/topologyspreadconstraint_test.go (hunk 1 above)
Comment (uthark) [reply pkg/descheduler/strategies/topologyspreadconstraint_test.go:6]: ``
`
./hack/verify-govet.sh
./hack/verify-spelling.sh
./hack/verify-gofmt.sh
!!! 'gofmt -s' needs to be run on the following files: 
./pkg/descheduler/strategies/topologyspreadconstraint_test.go
make: *** [Makefile:107: verify-gofmt] Error 1
```

**Review by damemi — COMMENTED** [review]
/approve

**Pull request merged**